src/transaction.ts

```diff
@@ -0,0 +1,270 @@
+import { Psbt, networks, address as laddress } from 'liquidjs-lib';
+import { estimateTxSize, UtxoInterface } from './wallet';
+
+export interface RecipientInterface {
+  value: number;
+  asset: string;
+  address: string;
+}
+
+export interface CoinSelectionResult {
+  selectedUtxos: UtxoInterface[];
+  changeOutputs: RecipientInterface[];
+}
+
+// define a type using to implement change's address strategy
+type ChangeAddressFromAssetGetter = (asset: string) => string | undefined;
+
+/**
+ * buildTx selects utxos among unspents to fill outputs' requirements,
+ * then it adds the associated inputs and outputs to the tx.
+ * finally it returns the new pset base64 encoded.
+ * @param psetBase64
+ * @param unspents
+ * @param outputs
+ * @param changeScriptByAsset
+ * @param addFee if true, estimate fee and add feeOutput to transaction (default = false)
+ * @param satsPerByte used for fee estimation (default = 0.1)
+ * @param network used for fee output (default = regtest)
+ */
+export function buildTx(
+  psetBase64: string,
+  unspents: UtxoInterface[],
+  recipients: RecipientInterface[],
+  changeAddressByAsset: ChangeAddressFromAssetGetter,
+  addFee: boolean = false,
+  satsPerByte: number = 0.1,
+  network: networks.Network = networks.regtest
+): string {
+  if (satsPerByte < 0.1) {
+    throw new Error('satsPerByte minimum value is 0.1');
+  }
+
+  if (recipients.length === 0) {
+    throw new Error(
+      'need a least one recipient output to build the transaction'
+    );
+  }
+
+  if (unspents.length === 0) {
+    throw new Error('need at least one unspent to fund the transaction');
+  }
+
+  const { selectedUtxos, changeOutputs } = greedyCoinSelection(
+    unspents,
+    recipients,
+    changeAddressByAsset
+  );
+
+  const inputs = selectedUtxos;
+
+  // if not fee, just add selected unspents as inputs and specified outputs + change outputs to pset
+  if (!addFee) {
+    const outs = recipients.concat(changeOutputs);
+    return addToTx(psetBase64, inputs, outs, network);
+  }
+
+  const pset = decodePset(psetBase64);
+  const nbInputs = pset.data.inputs.length + inputs.length;
+  const nbOutputs = pset.data.outputs.length + recipients.length + 1;
```

Comment:
what about the changes? We should include all of them, not just 1 (I suppose it refers to the LBTC change)
```suggestion
  const nbOutputs = pset.data.outputs.length + recipients.length + changeOutputs.length;
```